### PR TITLE
docs; license headers; bump sentry client version

### PR DESCRIPTION
**Pull request by yjbanov**
/cc @eseidelGoogle

### CODE REVIEW

**Comment by eseidelGoogle** [comment]
<3 . @Hixie is really the right reviewer.

File: lib/main.dart [hunk 1]
```diff
@@ -1,16 +1,46 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// This is a sample Flutter app that demonstrates to to catch various kinds
```
Comment (goderbauer) [inline lib/main.dart]: remove one "to"?

File: lib/main.dart [hunk 2]
```diff
@@ -1,16 +1,46 @@
...
+library crashy;
+
 import 'dart:async';
 import 'dart:isolate';
 
 import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter/services.dart';
+
+// This import the Dart Sentry.io client that sends crash reports to Sentry.io.
```
Comment (goderbauer) [inline lib/main.dart]: import -> imports?

File: lib/main.dart [hunk 3]
```diff
@@ -1,16 +1,46 @@
...
 import 'package:sentry/sentry.dart';
 
-// This file does not exist in the repository and is .gitignored. You have to
-// create one and add a `dsn` constant String containing your DSN value issued
-// by Sentry.io to your project.
+// This file does not exist in the repository and is .gitignored. In your local
+// clone of this repository, create this file and add a top-level [String]
+// constant containing the DSN value issued by Sentry.io to your project.
```
Comment (goderbauer) [inline lib/main.dart]: Should this be mentioned in the README so people know it upfront and not after the app fails to build?

Comment (yjbanov) [reply lib/main.dart]: Done.

**Comment by yjbanov** [comment]
Also:

- Added logic for omitting Sentry reports in dev mode.
- `FlutterError` handler now forwards errors to the zone handler, to avoid reporting failures in the reporting logic itself (FlutterError catches an error but crashes, which is caught by the Zone, which attempts to send a report the FlutterError crash).

PTAL.

**Pull request merged**